crates/ruff_python_semantic/src/analyze/class.rs

```diff
@@ -170,3 +171,58 @@ pub fn is_metaclass(class_def: &ast::StmtClassDef, semantic: &SemanticModel) ->
         (false, _) => IsMetaclass::No,
     }
 }
+
+/// Returns true if a class might generic.
+///
+/// A class is considered generic if at least one of its direct bases
+/// is subscripted with a `TypeVar`-like,
+/// or if it is defined using PEP 695 syntax.
+///
+/// This should only be used in stub context to avoid false positives and negatives.
+pub fn might_be_generic(class_def: &ast::StmtClassDef, semantic: &SemanticModel) -> bool {
+    if class_def.type_params.is_some() {
+        return true;
+    }
+
+    class_def.bases().iter().any(|base| {
+        let Expr::Subscript(ExprSubscript { slice, .. }) = base else {
+            return false;
+        };
+
+        let Expr::Tuple(ExprTuple { elts, .. }) = slice.as_ref() else {
+            return expr_might_be_typevar_like(slice, semantic);
+        };
+
+        elts.iter()
+            .any(|elt| expr_might_be_typevar_like(elt, semantic))
+    })
+}
+
+fn expr_might_be_typevar_like(expr: &Expr, semantic: &SemanticModel) -> bool {
+    is_known_typevar(expr, semantic) || expr_might_be_old_style_typevar_like(expr, semantic)
+}
+
+fn is_known_typevar(expr: &Expr, semantic: &SemanticModel) -> bool {
+    semantic.match_typing_expr(expr, "AnyStr")
+}
+
+fn expr_might_be_old_style_typevar_like(expr: &Expr, semantic: &SemanticModel) -> bool {
+    match expr {
+        Expr::Name(name) => might_be_old_style_typevar_like(name, semantic),
+
+        Expr::Starred(ExprStarred { value, .. }) => match value.as_ref() {
+            Expr::Name(name) => might_be_old_style_typevar_like(name, semantic),
+            _ => false,
+        },
+
+        _ => false,
+    }
+}
+
+fn might_be_old_style_typevar_like(name: &ExprName, semantic: &SemanticModel) -> bool {
+    let Some(binding) = semantic.only_binding(name).map(|id| semantic.binding(id)) else {
+        return false;
```

Comment:
this should return `true` because the binding _might_ refer to a TypeVar if there are no bindings (it could be imported via a `*` import) or multiple bindings